include/aws/io/io.h

```diff
@@ -250,6 +250,8 @@ enum aws_io_errors {
     AWS_IO_STREAM_SEEK_UNSUPPORTED,
     AWS_IO_STREAM_GET_LENGTH_UNSUPPORTED,
 
+    AWS_IO_TLS_ERROR_READ_FAILURE,
```

Comment:
Thanks, good catch. This is confusing. 

Debateable:  
For Windows: I have updated the error to `AWS_IO_TLS_ERROR_READ_FAILURE`, as `AWS_IO_TLS_ERROR_NEGOTIATION_FAILURE` is incorrect. We can see the logs to see the exact error that occurred.

For Apple: We can update to read failure, but since it has the concept of graceful vs. non-graceful shutdown, so `AWS_IO_TLS_ERROR_ALERT_RECEIVED` kinda makes sense. But also, that is the only place that we throw that error.

Reply:
I would also change the Apple code from ~AWS_IO_TLS_ERROR_ALERT_RECEIVED~ to `AWS_IO_TLS_ERROR_READ_FAILURE`.

I mean, it's the same situation of "this function returned any error other than WOULD_BLOCK"

Reply:
Thanks, updated.